tests/gen_sim_input.py

```diff
@@ -68,18 +68,34 @@ class RandExp(object):
 
   def __init__(self, num_unique_values, dist_param):
     self.num_unique_values = num_unique_values
-    self.lambda_ = dist_param or float(num_unique_values) / 5
-    # 1 - e^-lambda
-    self.one_minus_exp_lambda = 1 - math.exp(-self.lambda_)
+    self.lambd = dist_param or 1
 
   def __call__(self):
-    rand_in_cf = random.random()
-    # Val sampled from exp distr in [0,1] is CDF^{-1}(unif in [0,1))
-    rand_sample_in_01 = (
-        -math.log(1 - rand_in_cf * self.one_minus_exp_lambda) / self.lambda_)
-    # Scale up to num_unique_values and floor to integer
-    rand_val = int((rand_sample_in_01 * self.num_unique_values) + 1)
-    return rand_val
+    while True:
+      r = random.expovariate(self.lambd)
+      value = int(round(r))
+      # Rejection sampling to drop outputs outside [1, num_unique_values]
+      if 1 <= value <= self.num_unique_values:
+        break
+
+    return value  # true client value
+
+class RandZipf(object):
+  """Returns a value drawn from a Zipf distribution."""
+
+  def __init__(self, num_unique_values, dist_param):
+    self.num_unique_values = num_unique_values
+    self.alpha = dist_param or 1
+
+  def __call__(self):
+    while True:
+      r = random.paretovariate(self.alpha)  # Zipf is a discrete Pareto
```

Comment:
Because of the Zipf / Pareto issue, let's delete gen_sim_inputs.py, and add gen_counts.R --raw, which just expands the "partition" variable to a text file.